packages/core/src/components/graphs/gauge.ts

```diff
@@ -0,0 +1,316 @@
+		return data.find((d) => d.group === "value").value;
+	}
+
+	getValueRatio(): number {
+		const value = Tools.clamp(this.getValue(), 0, 100);
+		return value / 100;
+	}
+
+	getDelta(): number {
+		const data = this.model.getData();
+		const delta = data.find((d) => d.group === "delta");
+		return delta ? delta.value : null;
+	}
+
+	getArcRatio(): number {
+		const options = this.model.getOptions();
+		const type = Tools.getProperty(options, "gauge", "type");
+		const arcRatio = type === GaugeTypes.FULL ? 1 : 0.5;
+		if (arcRatio === undefined) {
+			throw new Error("Gauge chart arc type not compatible");
+		}
+		return arcRatio;
+	}
+
+	getArcSize(): number {
+		return this.getArcRatio() * Math.PI * 2;
+	}
+
+	getStartAngle(): number {
+		const arcSize = this.getArcSize();
+		if (arcSize === 2 * Math.PI) {
+			return 0;
+		}
+		return -arcSize / 2;
+	}
+
+	// use provided arrow direction or default to using the delta
+	getArrow(delta): string {
+		const options = this.model.getOptions();
+		const arrowDirection = Tools.getProperty(options, "gauge", "arrowDirection");
+
+		switch (arrowDirection) {
+			case ArrowDirections.UP:
+				return ARROW_UP;
+			case ArrowDirections.DOWN:
+				return ARROW_DOWN;
+			default:
+				return delta > 0 ? ARROW_UP : ARROW_DOWN;
+		}
+	}
+
+
+	render(animate = true) {
+		const self = this;
+		const svg = this.getContainerSVG();
+		const options = this.model.getOptions();
+		const value = this.getValue();
+		const valueRatio = this.getValueRatio();
+		const delta = this.getDelta();
+		const arcSize = this.getArcSize();
+		const arcType = Tools.getProperty(options, "gauge", "arcType");
+		const startAngle = this.getStartAngle();
+		const rotationAngle = valueRatio * arcSize;
+		const currentAngle = startAngle + rotationAngle;
+		const endAngle = startAngle + arcSize;
+
+		// Compute the outer radius needed
+		const radius = this.computeRadius();
+		const innerRadius = this.getInnerRadius();
+
+		// Sizing and positions relative to the radius
+		const arrowSize = radius / 8;
+		const valueFontSize = radius / 2.5;
+		const deltaFontSize = radius / 8;
+		const distanceBetweenNumbers = 10;
+		const numbersYPosition =
+			arcType === GaugeTypes.SEMI
+				? -(deltaFontSize + distanceBetweenNumbers)
+				: 0;
+
+		this.backgroundArc = arc()
+			.innerRadius(innerRadius)
+			.outerRadius(radius)
+			.startAngle(currentAngle)
+			.endAngle(endAngle);
+
+		this.arc = arc()
+			.innerRadius(innerRadius)
+			.outerRadius(radius)
+			.startAngle(startAngle)
+			.endAngle(currentAngle);
+
+		// draw the container
+		DOMUtils.appendOrSelect(svg, "path.arc-background")
+			.attr("d", this.backgroundArc)
+			.attr("role", Roles.GROUP);
+
+
+		// Add data arc
+		const arcValue = svg.selectAll("path.arc-foreground")
+			.data([value]);
+
+		arcValue
+			.enter()
+			.append("path")
+			.attr("class", "arc-foreground")
+			.merge(arcValue)
+			.attr("d", this.arc)
+			.attr("fill", Tools.getProperty(options, "gauge", "fillColor"))
```

Comment:
shouldn't this be set through `colorScale`? https://carbon-design-system.github.io/carbon-charts/?path=/story/tutorials--tabular-data-format